File: test/org/cerion/symcalc/expression/function/list/IdentityMatrixTest.java
package org.cerion.symcalc.expression.function.list;

import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.ListExpr;
import org.cerion.symcalc.expression.constant.Pi;
import org.cerion.symcalc.expression.number.IntegerNum;
import org.cerion.symcalc.expression.number.RealNum;
import org.junit.Assert;
import org.junit.Test;

public class IdentityMatrixTest {

    @Test
    public void validate() {
        // Non-integers
        Assert.assertEquals(Expr.ExprType.ERROR, new IdentityMatrix(RealNum.Companion.create(2.34)).eval().getType());
        Assert.assertEquals(Expr.ExprType.ERROR, new IdentityMatrix(new Pi()).eval().getType());

        // Zero
        Assert.assertEquals(Expr.ExprType.ERROR, new IdentityMatrix(IntegerNum.ZERO).eval().getType());

        // Negative
        Assert.assertEquals(Expr.ExprType.ERROR, new IdentityMatrix(new IntegerNum(-10)).eval().getType());
    }

    @Test
    public void basic() {
        Expr e = new IdentityMatrix(new IntegerNum(3));
        ListExpr expected = new ListExpr(
                new ListExpr(IntegerNum.ONE, IntegerNum.ZERO, IntegerNum.ZERO),
                new ListExpr(IntegerNum.ZERO, IntegerNum.ONE, IntegerNum.ZERO),
                new ListExpr(IntegerNum.ZERO, IntegerNum.ZERO, IntegerNum.ONE)
        );

        Assert.assertEquals(expected, e.eval());
    }
}

File: test/expression/function/list/IdentityMatrixTest.kt
package org.cerion.symcalc.expression.function.list

import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.expression.ListExpr
import org.cerion.symcalc.expression.constant.Pi
import org.cerion.symcalc.expression.number.Integer
import org.cerion.symcalc.expression.number.RealDouble
import org.junit.Assert
import org.junit.Test

class IdentityMatrixTest {

    @Test
    fun validate() {
        // Non-integers
        Assert.assertEquals(Expr.ExprType.ERROR, IdentityMatrix(RealDouble(2.34)).eval().type)
        Assert.assertEquals(Expr.ExprType.ERROR, IdentityMatrix(Pi()).eval().type)

        // Zero
        Assert.assertEquals(Expr.ExprType.ERROR, IdentityMatrix(Integer.ZERO).eval().type)

        // Negative
        Assert.assertEquals(Expr.ExprType.ERROR, IdentityMatrix(Integer(-10)).eval().type)
    }

    @Test
    fun basic() {
        val e = IdentityMatrix(Integer(3))
        val expected = ListExpr(
                ListExpr(Integer.ONE, Integer.ZERO, Integer.ZERO),
                ListExpr(Integer.ZERO, Integer.ONE, Integer.ZERO),
                ListExpr(Integer.ZERO, Integer.ZERO, Integer.ONE)
        )

        Assert.assertEquals(expected, e.eval())
    }
}